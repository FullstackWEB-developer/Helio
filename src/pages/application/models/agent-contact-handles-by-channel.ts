import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
