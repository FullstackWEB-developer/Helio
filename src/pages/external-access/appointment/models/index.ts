export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
