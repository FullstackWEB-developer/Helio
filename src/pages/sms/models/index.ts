export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
