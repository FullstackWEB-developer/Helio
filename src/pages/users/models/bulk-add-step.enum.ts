export enum BulkAddStep {
    Selection = 1,
    RolePicking,
    ProviderMapping,
    Review
}