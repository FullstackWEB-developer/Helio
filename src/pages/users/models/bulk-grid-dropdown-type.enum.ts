export enum BulkGridDropdownType {
    Role,
    ProviderMapping
}