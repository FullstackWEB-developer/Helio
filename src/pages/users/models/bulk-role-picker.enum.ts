export enum BulkRolePicker {
    Group,
    Individual
}