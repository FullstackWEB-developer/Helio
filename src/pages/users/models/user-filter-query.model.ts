export interface UserQueryFilter {
    searchText?: string;
    jobTitle?: string;
    departments?: string;
    roles?: string;
    invitationStatuses?: string;
    statuses?: string;
    rolesUnassigned?: string;
}