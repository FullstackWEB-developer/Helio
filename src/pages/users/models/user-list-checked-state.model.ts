import {CheckboxCheckEvent} from '@components/checkbox/checkbox';
import {UserDetailStatus, UserInvitationStatus} from '@shared/models';

export interface UserListCheckedState {
    checkboxCheckEvent: CheckboxCheckEvent;
    userStatus: UserDetailStatus;
    userInvitationStatus?: UserInvitationStatus;
    userEmail: string;
}