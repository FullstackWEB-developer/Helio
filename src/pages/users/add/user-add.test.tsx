import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {fireEvent, render, wait, waitFor} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import UserAdd from './user-add';
import { act } from 'react-dom/test-utils';
import Router from "react-router";
import Api from '@shared/services/api';
import { PagedList, UserActiveDirectory } from '@shared/models';
jest.setTimeout(30000);
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useParams: jest.fn().mockReturnValue({ type: "ticket-department" }),
  }))
describe("UserAdd tests", () => {
    
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {
            providerList: [
                {
                    firstName: "string",
                    specialty: "string",
                    homeDepartment: "string",
                    specialtyId: 0,
                    schedulingName: "string",
                    providerTypeId: "string",
                    billable: false,
                    displayName: "string",
                    ansiNameCode: "string",
                    lastName: "string",
                    id: 0,
                    providerUserName: "string",
                    ansiSpecialtyCode: "string",
                    hideInPortal: true,
                    sex: "string",
                    entityType: 0,
                    providerType: "string",
                    createEncounterOnCheckin: false
                },{
                    firstName: "string",
                    specialty: "string",
                    homeDepartment: "string",
                    specialtyId: 1,
                    schedulingName: "string",
                    providerTypeId: "string",
                    billable: false,
                    displayName: "string",
                    ansiNameCode: "string",
                    lastName: "string",
                    id: 1,
                    providerUserName: "string",
                    ansiSpecialtyCode: "string",
                    hideInPortal: true,
                    sex: "string",
                    entityType: 0,
                    providerType: "string",
                    createEncounterOnCheckin: false
                }
            ],
            roleList: [
                {
                    name: "string0",
                    description: "string0",
                },{
                    name: "string1",
                    description: "string1",
                }
            ]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        snackbarState: {
            messages: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders UserAdd correctly", async () => {
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <UserAdd/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders UserAdd with data correctly", async () => {
        jest.spyOn(Api, 'post').mockImplementation(() => {
            return new Promise((resolve) => {
                resolve({})
            })
        });
        jest.spyOn(Api, 'get').mockImplementation((url) => {
            if (url.includes('external-users')) {
                return new Promise((resolve) => {
                    resolve({
                        data: {
                            page: 1,
                            pageSize: 10,
                            totalPages: 1,
                            totalCount: 1,
                            results: [
                                {
                                    azureId: "190981e5-df9a-44b8-b534-da788eba29a3",
                                    displayName: "Test",
                                    description: "Empty",
                                    mail: "test@gmail.com",
                                }
                            ]
                    } as PagedList<UserActiveDirectory>
                    })
                })
            }else {
                return new Promise((resolve) => {
                    resolve({})
                })
            }
        });
        await act(async () => {
            const {getByTestId} = render(<TestWrapper mockState={mockState}>
                <UserAdd/>
            </TestWrapper>);
            await waitFor(() => {
                fireEvent.click(getByTestId('email'));
                fireEvent.input(getByTestId('users.add_section.personalize_message_placeholder'), { target: {value: "Test"}});
                fireEvent.input(getByTestId('email'), { target: {value: "Test"}});
                setTimeout(() => {}, 5000);
                fireEvent.change(getByTestId('role'), { target: {value: "string0"}});
                fireEvent.click(getByTestId('add-user'));
                fireEvent.click(getByTestId('users.add_section.add_button'));
                
            });
        });
        expect(Api.get).toHaveBeenCalledTimes(2);
        expect(Api.post).toHaveBeenCalledTimes(0);
    });
})