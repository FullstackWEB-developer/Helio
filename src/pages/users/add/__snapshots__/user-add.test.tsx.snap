// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAdd tests renders UserAdd correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col w-full px-6 py-8 overflow-y-auto user-add"
  >
    <h5>
      users.add_section.title
    </h5>
    <span
      class="mt-10"
    >
      <a
        class="link"
        href="/users/bulk"
      />
    </span>
    <div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-row items-center mt-3.5"
        >
          <div
            class="mr-8 w-80"
          >
            <div
              class="flex flex-col h-20 input-group"
            >
              <div
                class="input-group-container flex flex-wrap items-stretch w-full relative  "
              >
                <input
                  autocomplete="off"
                  class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                  data-test-id="email"
                  data-testid="email"
                  name="email"
                  placeholder=""
                  type="text"
                  value=""
                />
                <label
                  class="absolute truncate  body2 "
                >
                  users.add_section.name_or_email
                </label>
              </div>
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="select-wrapper relative w-full flex flex-col h-20 "
            >
              <div
                class="select relative flex flex-col "
              >
                <input
                  autocomplete="on"
                  autocorrect="on"
                  class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
                  data-testid="role"
                  name="role"
                  spellcheck="true"
                  type="text"
                  value=""
                />
                <label
                  class="select-label absolute truncate"
                >
                  <span
                    class="select-label-span body2"
                  >
                    users.add_section.role
                  </span>
                </label>
                <div
                  class="absolute pt-4 right-4 "
                >
                  <div
                    class="flex flex-row items-start"
                  >
                    <div
                      class="cursor-pointer"
                    >
                      <div
                        class=""
                      >
                        <svg
                          class="icon-medium "
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="none"
                            height="24"
                            width="24"
                          />
                          <path
                            class="select-arrow-fill"
                            d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                            transform="translate(5 16.273) rotate(-90)"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="absolute py-2 options"
                >
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-string0"
                        data-testid="select-cell-text-string0"
                      >
                        string0
                      </div>
                    </div>
                  </div>
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-string1"
                        data-testid="select-cell-text-string1"
                      >
                        string1
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-row items-center w-80"
          >
            <div
              class="select-wrapper relative w-full flex flex-col h-20 "
            >
              <div
                class="select relative flex flex-col "
              >
                <input
                  autocomplete="on"
                  autocorrect="on"
                  class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
                  data-testid="providerId"
                  name="providerId"
                  spellcheck="true"
                  type="text"
                  value=""
                />
                <label
                  class="select-label absolute truncate"
                >
                  <span
                    class="select-label-span body2"
                  >
                    users.ehr_provider_mapping
                  </span>
                </label>
                <div
                  class="absolute pt-4 right-4 "
                >
                  <div
                    class="flex flex-row items-start"
                  >
                    <div
                      class="cursor-pointer"
                    >
                      <div
                        class=""
                      >
                        <svg
                          class="icon-medium "
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="none"
                            height="24"
                            width="24"
                          />
                          <path
                            class="select-arrow-fill"
                            d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                            transform="translate(5 16.273) rotate(-90)"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="absolute py-2 options"
                >
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-string string"
                        data-testid="select-cell-text-string string"
                      >
                        string string
                      </div>
                    </div>
                  </div>
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-string string"
                        data-testid="select-cell-text-string string"
                      >
                        string string
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="h-6 max-h-6 pl-4  body3 pt-1 truncate"
              >
                <span
                  class="assistive-text-color-inactive"
                >
                  users.ehr_provider_mapping_assistive_text
                </span>
              </div>
            </div>
          </div>
          <div
            class="pb-6 ml-2"
          >
            <div
              class="provider-mapping-tool-tip"
              data-testid="toggle"
            >
              <div
                class=""
              >
                <svg
                  class="cursor-pointer icon "
                  height="16"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="https://www.w3.org/2000/svg"
                >
                  <g
                    transform="translate(-16.988 -15.519)"
                  >
                    <circle
                      class="warning-icon"
                      cx="24.988"
                      cy="23.519"
                      r="8"
                    />
                    <rect
                      fill="#fff"
                      height="5"
                      width="2"
                      x="23.988"
                      y="19.519"
                    />
                    <rect
                      fill="#fff"
                      height="2"
                      width="2"
                      x="23.988"
                      y="25.519"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
          <div
            class="pb-6 ml-8"
          >
            <button
              class="medium-button"
              data-testid="add-user"
              disabled=""
              type="button"
            >
              <div
                class="flex flex-row justify-center items-center space-x-2"
              >
                <div>
                  common.add
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-8"
    >
      <label
        class="ml-4 subtitle2"
      >
        users.add_section.personalize_message_title
      </label>
      <div
        class="flex flex-row justify-between"
      >
        <label
          class="block subtitle2 pb-2"
        />
      </div>
      <div
        class="w-full h-full"
      >
        <div
          class="body2 pl-4 pt-2 relative bg-white border"
        >
          <textarea
            class="shadow-none invitation-message-text-area resize-none mt-1"
            data-testid="users.add_section.personalize_message_placeholder"
            placeholder="users.add_section.personalize_message_placeholder"
            rows="3"
            wrap="hard"
          />
        </div>
      </div>
    </div>
    <div
      class="mt-10"
    >
      <button
        class="medium-button"
        data-testid="users.add_section.add_button"
        disabled=""
        type="button"
      >
        <div
          class="flex flex-row justify-center items-center space-x-2"
        >
          <div>
            users.add_section.add_button
          </div>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
